Add section on accounts and keys (#99)
clearhaus/gateway-api-docs: source/index.html.md

key.sign(OpenSSL::Digest.new('sha256'), body).unpack('H*').first
```

## Accounts and keys

There are two types of accounts:

* A merchant account belongs to a merchant.

> * A partner account belongs to you, i.e. a partner that is tech-wise integrated

Reviewer: ```suggestion
* A partner account belongs to you, i.e. a partner being tech-wise integrated
```

Author: ba0261d

  into the transaction gateway.

Merchant accounts can process transactions whereas partner accounts cannot.

An API key (UUIDv4) points to an account (either a merchant account or a partner
account). An account can have multiple API keys pointing to it; this allows for

> easy rolling of API keys. A merchant account usually has one API key.

Reviewer 2: ```suggestion
easy rolling of API keys. A merchant account usually uses one API key.
```

A very minor comment that you may discard at your whim :)

Author: ba0261d

API keys can have an expire date but usually do not. API keys can be enabled or
disabled. For merchant accounts, these details can be observed in the Clearhaus
Dashboard.

> An API key can be a signing API key; a signing API key has a verification key

Reviewer: ```suggestion
An API key can be a signing API key, meaning that it has a verification key for
```

Author: ba0261d

> that can verify signatures. This verification key is the public part of an RSA

Reviewer: ```suggestion
verifying request signatures. This verification key is the public part of an RSA
```

Author: ba0261d

> An API key can be a signing API key; a signing API key has a verification key
> that can verify signatures. This verification key is the public part of an RSA
> key pair; the RSA key pair is generated by the partner. The public part, the
> verification key, is communicated to Clearhaus during the technical integration.
> 
> Only partner accounts have signing API keys (because Clearhaus puts trust in
> partners to process transactions correctly). Thus, in the `Signature` header,
> the partner's signing API key is supplied together with the hex signature; this
> hex signature is created using the private part of the RSA key pair such that
> the hex signature can be verified using the verification key that belongs to
> that signing API key.

Reviewer 2: ```suggestion
A partner account API key can be a signing API key; a signing API key has a
verification key that can verify signatures. This verification key is the
public part of an RSA key pair; the RSA key pair is generated by the partner.
The public part, the verification key, is communicated to Clearhaus during the
technical integration.  This signing API key is used to sign requests, as
described in the [Signing requests](#signing-requests) section.
```

I suggest we move all description of signing requests to that section. (Incorporate or ignore at your leisure :+1:)

For:
> This signing API key is used

I considered changing it to be:
> This signing API key can be used

But I feel like this implies that signing is optional.

Author: I thought about it shortly, but (a) wanted it resolved fast, and (b) the signing section is "technical only", and perhaps it's nice to keep it that way — WDYT?

Reviewer 2: I'd prefer that sections are self-contained and don't raise further questions (e.g. how do I sign requests), but I see the point of it being technical only. Just commit it as it is, I think it would be better done in a larger rework :+1:

Reviewer 3: I prefer the shorter version and then linking to #signing-request. But "how do I sign requests?" is the most frequent questions PSPs have. So at bit more verbosity might be good. Either way :+1:

Author: Is ba0261d acceptable?
